### PR TITLE
ci: Add teardown script

### LINKED ISSUE #83: Add teardown script
Add teardown script to collect logs in case a failure occurs.

### PR DESCRIPTION
**Pull request by chavafg**
Teardown script will collect all logs from the runtime,
proxy and shim.

Fixes #83.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

File: .ci/teardown.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+proxy_log_filename="kata-proxy.log"
+proxy_log_path="${log_copy_dest}/${proxy_log_filename}"
+proxy_log_prefix="kata-proxy_"
+
+shim_log_filename="kata-shim.log"
+shim_log_path="${log_copy_dest}/${shim_log_filename}"
+shim_log_prefix="kata-shim_"
+
+crio_log_filename="crio.log"
```
Comment (grahamwhaley) [inline .ci/teardown.sh]: wrt the commit message - it also appears to collect from CRIO and docker - do you want to note that as well?

Comment (chavafg) [reply .ci/teardown.sh]: ohh right , will update with that info :)

**Comment by GabyCT** [comment]
lgtm

File: .ci/teardown.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		"${shim_log_prefix}" \
+		"${crio_log_prefix}"
+		"${docker_log_prefix}"
+	do
+		gzip -9 "$prefix"*
+	done
+
+	popd
+else
+	echo "Clear Containers Runtime Log:"
```
Comment (jodh-intel) [inline .ci/teardown.sh]: `s/Clear Containers/Kata Containers/g` ;)

File: .ci/teardown.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	journalctl --no-pager -t kata-runtime-cc
```
Comment (jodh-intel) [inline .ci/teardown.sh]: Again, we need to handle `kata-runtime-runv` I think.

File: .ci/teardown.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+docker_log_filename="docker.log"
+docker_log_path="${log_copy_dest}/${docker_log_filename}"
+docker_log_prefix="docker_"
+
+# Copy log files if a destination path is provided, otherwise simply
+# display them.
+if [ ${log_copy_dest} ]; then
+	# Create the log files
+	journalctl --no-pager -t kata-runtime-cc > "${runtime_log_path}"
```
Comment (jodh-intel) [inline .ci/teardown.sh]: Strictly, this should be looking for the runtime name that's been configured for docker/k8s. But for simplicity, how about we create two runtime log files, one for `kata-runtime-cc` and another for `kata-runtime-runv`?

Comment (chavafg) [reply .ci/teardown.sh]: Agree, lets create two log files now and once we move on with the https://github.com/kata-containers/runtime/issues/33, we can modify it.

**Comment by chavafg** [comment]
@jodh-intel changes applied

File: .ci/teardown.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+		"${shim_log_prefix}" \
+		"${crio_log_prefix}" \
+		"${docker_log_prefix}"
+	do
+		gzip -9 "$prefix"*
+	done
+
+	popd
+else
+	echo "Kata Containers Runtime (Runv) Log:"
```
Comment (jodh-intel) [inline .ci/teardown.sh]: I think the first two `echo`'s here are the wrong way round?

Comment (chavafg) [reply .ci/teardown.sh]: ohh true, seems like I am still sleepy... Fixed :)

**Comment by jodh-intel** [comment]
Thanks @chavafg.

lgtm

**Pull request merged**